vivisect/parsers/elf.py

```diff
@@ -531,7 +532,12 @@ def loadElfIntoWorkspace(vw, elf, filename=None, baseaddr=None):
                     while ugly, this does cover a couple nice use cases like pointer tables/arrays of pointers being present.
                     '''
                     if not valu:
-                        new_pointers.append((sva, valu, symname))
+                        # do a double check to make sure we can even make a pointer this large
+                        # because some relocations like __FRAME_END__ might end up short
+                        psize = vw.getPointerSize()
+                        byts = vw.readMemory(sva, psize)
+                        if len(byts) == psize:
```

Comment:
So this is a special case with how the `__FRAME_END__` relocation gets handled. At least on these binaries, it ends up being a 4 byte string of null bytes at the end of the map.

But to the main question, yes. It can. With how `readMemory` works: `return mbytes[offset:offset+size]`, you can end up with situations like these:
```
>>> x = [1,2,3,4,5,6,7,8,9]
>>> x[7:15]
[8, 9]
```

I'm tempted to say that I would want it to scream if it is forced to do a short read, but that's a super fundamental change to how readMemory works, so I'm hesitant to put that in a minor release.

Reply:
gotcha.  we should probably do the math and raise an exception if we read past the end of a map.  or check to see if we have an adjacent map we can continue reading from and return what's requested, assuming permissions allow?

Reply:
Assuming permissions. I think we've talked about something like this with the metamap stuff we've briefly discussed? But I figured that was it's own behemoth of changes that might upset some conventions, so I left it out of this PR so that the base APIs still work the same.

Reply:
yep.  the cross-map thing was what we discussed...
but shouldn't we raise an exception if we run past the end of a map?  that's something that seems like a problem we should solve right now instead of checking the length after the fact.

Reply:
It's a fundamental enough change that I think it deserves it own PR (rather than trying to sneak it in here). 

I like the idea, I'm just trying to get a bugfix release for the capa guys out as quickly as possible, while not causing too much disruption to their workflows. So maybe we can slate that for a 1.0.6 or a 1.1.0, with 1.0.5 being a smaller bugfix release? I'm open to suggestions.

Reply:
makes sense.